src/client.rs

```diff
@@ -0,0 +1,208 @@
+use crate::{
+    meta::{List, ListOptions},
+    rest::{self, ClientConfig},
+};
+use anyhow::{Error, Result};
+use reqwest::{Method, RequestBuilder};
+use serde::{de::DeserializeOwned, Serialize};
+use serde_with::*;
+
+pub struct Client {
+    pub rest: rest::Client,
```

Comment:
Naming two separate structs `Client`, even if both are internal, is clunky.
I there is no need for both of them.